app/api/preference_api.py

```diff
@@ -148,14 +149,17 @@ def get_txn_data():
     next_cursor = None
     txn_sets_map = {}
     more = True
+    total_prefs = 0
 
-    while(more):
+    # if True:
+    while(more and total_prefs < 8000):
```

Comment:
In current master this looks over Transactions rather than individual Preferences and limit of 1000 Transactions is used.